osu.Game/Overlays/News/Sidebar/MonthSection.cs

```diff
@@ -0,0 +1,197 @@
+                AutoSizeAxes = Axes.Y;
+                Child = text = new TextFlowContainer(t => t.Font = OsuFont.GetFont(size: 12))
+                {
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Text = post.Title
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OverlayColourProvider colourProvider, GameHost host)
+            {
+                IdleColour = colourProvider.Light2;
+                HoverColour = colourProvider.Light1;
+
+                TooltipText = "view in browser";
+                Action = () => host.OpenUrlExternally("https://osu.ppy.sh/home/news/" + post.Slug);
+            }
+        }
+
+        private class PostsContainer : Container
+        {
+            public readonly BindableBool IsOpen = new BindableBool();
+
+            protected override Container<Drawable> Content => content;
+
+            private readonly FillFlowContainer content;
+
+            public PostsContainer()
+            {
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+                InternalChild = content = new FillFlowContainer
+                {
+                    Margin = new MarginPadding { Top = 5 },
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Direction = FillDirection.Vertical,
+                    Spacing = new Vector2(0, 5)
+                };
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                IsOpen.BindValueChanged(_ => updateState(), true);
+
+                // First state change should be instant.
+                FinishTransforms(true);
+            }
+
+            private void updateState()
+            {
+                ClearTransforms(true);
+
+                if (IsOpen.Value)
+                {
+                    AutoSizeAxes = Axes.Y;
+                    content.FadeIn(animation_duration, Easing.OutQuint);
+                }
+                else
+                {
+                    AutoSizeAxes = Axes.None;
+                    this.ResizeHeightTo(0, animation_duration, Easing.OutQuint);
+
+                    content.FadeOut(animation_duration, Easing.OutQuint);
+                }
+            }
+
+            private bool autoSizeTransitionApplied;
+
+            // Workaround to allow the dropdown to be opened immediately since FinishTransforms doesn't work for AutoSize{Duration,Easing}.
+            protected override void UpdateAfterAutoSize()
+            {
+                base.UpdateAfterAutoSize();
+
+                if (!autoSizeTransitionApplied)
+                {
+                    AutoSizeDuration = animation_duration;
+                    AutoSizeEasing = Easing.OutQuint;
+
+                    autoSizeTransitionApplied = true;
+                }
+            }
```

Comment:
I'm not sure what this is trying to achieve, but it looks fine without this logic?

Reply:
Makes the posts container autosize immediately on first frame while applying autosize transforms on state changes afterwards.

For some reason this can't be done with a scheduled `FinishTransforms`, although I got to admit I haven't checked thoroughly why, will do in a bit.

Reply:
I'm fine with removing. It's not like if this looks bad if the section expand animation plays out on first load instead of instantly.

The reason this can't be done with a schedule is that schedules (both normal, and after children) run too late. [Autosize is the last thing in a subtree update](https://github.com/ppy/osu-framework/blob/5a002b4eeb9598954b26fb86cc93123933d4f52c/osu.Framework/Graphics/Containers/CompositeDrawable.cs#L888).

Reply:
I did actually mean a 2-frame scheduling but yeah, what's the point at this point, let the animation play and call it a day.